kornia/color/hls.py

```diff
@@ -148,4 +148,6 @@ def rgb_to_hls(image):
 
     h: torch.Tensor = (60 * hi) / 360  # hue
 
-    return torch.stack([h, l, s], dim=-3)
+    image = torch.stack([h, l, s], dim=-3)
+    image[torch.isnan(image)] = 0
```

Comment:
if is easier to add an epsilon value here:

```python
eps: float = 1e-8
denom: torch.Tensor = maxc + minc + eps
s: torch.Tensor = torch.where(l < 0.5, deltac / denom, deltac / (2 - denom))  # saturation
```

Reply:
Nope. When I will add `epsilon = 1e-9` some pixels in return value will be different to OpenCV results.
In the code above with uint8 image difference in some pixel will be `255.`
```
def test(img):
    img = K.rgb_to_hls(img.float() / 255)
    img[0] *= 360

    img[0] /= 2
    img[1:] *= 255

    return img.round().clamp(0, 255).type(torch.uint8)

np.random.seed(0)

img = np.random.randint(0, 256, [1000, 1000, 3], np.uint8)

cv_img = cv.cvtColor(img, cv.COLOR_RGB2HLS)

k_img = torch.from_numpy(np.transpose(img, [2, 0, 1]))
k_img = test(k_img)
k_img = np.transpose(k_img.numpy(), [1, 2, 0])

print(np.abs(cv_img.astype(int) - k_img.astype(int)).max())
```

Reply:
ok, I will do this soon

Reply:
@Dipet have you tried with images in floating point? opencv usually has slightly different implementation for different data types. Our scope is training meaning that we usually assume data coming in floating point precision

Reply:
@edgarriba OpenCV uses epsilon for this conversion https://github.com/opencv/opencv/blob/a165f5557974c7ba32e5d6c966ec053d2bf932cc/modules/imgproc/src/color_hsv.simd.hpp#L688
So, I will implement it like in OpenCV and I will think how implement conversion in uint8.